src/services/email-line-item-delivery.js

```diff
@@ -28,93 +29,105 @@ module.exports = {
    * @param {Date} params.date
    * @param {object} [params.imageOptions] Imgix options to use in the creative image url.
    * @param {object} [params.advertiserLogoOptions] Imgix options to use in the creative image url.
+   * @param {object} [params.opts] Generic options see campaign-delivery.js for potential examples
    */
   async findFor({
     placementId,
     date,
     imageOptions,
     advertiserLogoOptions,
+    opts,
```

Comment:
I know that you are using the `parseOption` util when accessing this from the email placement router, which will always return an object, however if this method is called from somewhere else this will error because `opts` will be undefined. Change this to either
- set a default value for opts on L39 (`opts = {},`)
- Use getAsObject or another util to ensure that `opts` is an object before accessing a property
- fake it (`const o = opts || {}; const limit = o.n ? parseLimit(o.n) : 1` or something)